### PR TITLE
Feature/35017 endpoint retrieve task impl

### PR DESCRIPTION
**Pull request by marijnritense**

### CODE REVIEW

File: form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/handler/FormFlowStepTypeFormHandler.kt [hunk 1]
```diff
@@ -60,9 +60,8 @@ class FormFlowStepTypeFormHandler(
         additionalProperties: Map<String, Any>
     ) {
 
-        // TODO: Make sure the FormFlowInstance.additionalProperties contain the 'documentId' and the 'taskInstanceId'
         val documentId = additionalProperties["documentId"] as String?
-        val taskInstanceId = additionalProperties["taskInstanceId"] as String?
+        val taskInstanceId = additionalProperties["taskId"] as String?
```
Comment (Klaas-Ritense) [inline form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/handler/FormFlowStepTypeFormHandler.kt:64]: The FormFlowCreateTaskEventHandler shows that it should be 'taskInstanceId'

File: form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/security/ValtimoFormFlowHttpSecurityConfigurer.kt [hunk 2]
```diff
@@ -27,9 +27,15 @@ class ValtimoFormFlowHttpSecurityConfigurer: HttpSecurityConfigurer {
...
-                .antMatchers(HttpMethod.GET, "/api/form-flow/{formFlowInstanceId}").hasAuthority(USER)
+                .antMatchers(HttpMethod.GET, "/api/form-flow/{formFlowInstanceId}")
+                .hasAuthority(USER)
                 .antMatchers(HttpMethod.POST, "/api/form-flow/{formFlowId}/step/{stepInstanceId}").hasAuthority(USER)
-                .antMatchers(HttpMethod.GET, "/api/process-link/form-flow-definition").hasAuthority(AuthoritiesConstants.ADMIN)
+                .antMatchers(HttpMethod.GET, "/api/process-link/form-flow-definition")
+                .hasAuthority(AuthoritiesConstants.ADMIN)
+                .antMatchers(HttpMethod.GET, "/api/form-flow/{formFlowId}")
+                .hasAuthority(AuthoritiesConstants.USER)
+                .antMatchers(HttpMethod.POST, "/api/form-flow/{formFlowId}/step/{stepInstanceId}")
```
Comment (Klaas-Ritense) [inline form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/security/ValtimoFormFlowHttpSecurityConfigurer.kt:37]: This list contain some duplicates

File: form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/web/rest/dto/FormTypeProperties.kt [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.formflow.web.rest.dto
+
+import com.fasterxml.jackson.databind.JsonNode
+
+class FormTypeProperties (
```
Comment (Klaas-Ritense) [inline form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/web/rest/dto/FormTypeProperties.kt:21]: This class already exists in the .handler. package

File: form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/web/rest/FormFlowResource.kt [hunk 4]
```diff
@@ -94,4 +113,19 @@ class FormFlowResource(
             )
         )
     }
+    private fun getStepDto(): FormFlow {
+        return FormFlow(
+            UUID.randomUUID(),
+            FormFlowStep(
+                UUID.randomUUID(),
+                "form",
+                FormTypeProperties(
```
[the remark this answers is not in the document]
Comment (ivo-ritense) [reply form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/web/rest/FormFlowResource.kt]: This was a merge gone wrong. Method has been removed.

File: form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/web/rest/dto/FormFlowStep.kt [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ritense.valtimo.formflow.web.rest.dto
+
+import java.util.UUID
+
+class FormFlowStep(
```
[the remark this answers is not in the document]
Comment (ivo-ritense) [reply form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/web/rest/dto/FormFlowStep.kt]: Fixed

File: form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/web/rest/dto/FormTypeProperties.kt (hunk 3 above)
Comment (ivo-ritense) [reply form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/web/rest/dto/FormTypeProperties.kt]: Fixed

File: form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/handler/FormFlowStepTypeFormHandler.kt (hunk 1 above)
Comment (ivo-ritense) [reply form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/handler/FormFlowStepTypeFormHandler.kt]: you're right

File: form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/security/ValtimoFormFlowHttpSecurityConfigurer.kt (hunk 2 above)
Comment (ivo-ritense) [reply form-flow-valtimo/src/main/kotlin/com/ritense/valtimo/formflow/security/ValtimoFormFlowHttpSecurityConfigurer.kt:33]: Merge issues i think, fixed.

**Review by Klaas-Ritense — APPROVED** [review]

**Pull request merged**